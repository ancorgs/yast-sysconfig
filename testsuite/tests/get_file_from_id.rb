# encoding: utf-8

# Sysconfig module - testsuite
#
# get_file_from_id tests
#
# testedfiles: Sysconfig.ycp
#
# $Id$
#
module Yast
  class GetFileFromIdClient < Client
    def main
      Yast.import "Testsuite"
      Yast.import "Pkg" # override packamanager
      Yast.import "Sysconfig"

      Testsuite.Test(lambda { Sysconfig.get_file_from_id(nil) }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.get_file_from_id("") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.get_file_from_id("var") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.get_file_from_id("var$file") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda { Sysconfig.get_file_from_id("var$") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.get_file_from_id("$file") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.get_file_from_id("var$x$file") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda { Sysconfig.get_file_from_id("$$varx$file") }, [
        {},
        {},
        {}
      ], nil)

      nil
    end
  end
end

Yast::GetFileFromIdClient.new.main
