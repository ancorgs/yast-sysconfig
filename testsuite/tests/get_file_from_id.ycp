// Sysconfig module - testsuite
//
// get_file_from_id tests
//
// $Id$
//

{

include "testsuite.ycp";

import "Pkg";   // override packamanager

import "Mode";		// mode is imported in Product
Mode::config = true;	// package manager is initialized in Product constructor when Mode::config is false !!
import "Sysconfig";

TEST( ``(Sysconfig::get_file_from_id(nil) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_file_from_id("") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_file_from_id("var") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_file_from_id("var$file") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_file_from_id("var$") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_file_from_id("$file") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_file_from_id("var$x$file") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_file_from_id("$$varx$file") ), [ $[], $[], $[] ], nil );

}

