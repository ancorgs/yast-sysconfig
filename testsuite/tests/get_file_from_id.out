Return	
Return	
Return	
Return	file
Return	
Return	file
Return	x$file
Return	$varx$file
